rpc/v1/publish_test.go

```diff
@@ -0,0 +1,206 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+
+	"github.com/matheusmosca/walrus/domain/entities"
+	"github.com/matheusmosca/walrus/domain/usecases"
+	"github.com/matheusmosca/walrus/domain/vos"
+	pb "github.com/matheusmosca/walrus/proto"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+func TestPublish(t *testing.T) {
+	type args struct {
+		ctx     context.Context
+		message vos.Message
+	}
+
+	type fields struct {
+		topicName vos.TopicName
+	}
+
+	type grpcError struct {
+		code codes.Code
+		msg  error
+	}
+
+	testCases := []struct {
+		name    string
+		args    args
+		fields  fields
+		wantErr *grpcError
```

Comment:
Could your replace this wantErr with this? You can remove grpcError as well
```suggestion
		wantErr error
                wantCode codes.Code
```